src/Go.CodeAnalysis/Text/SnapshotBase.cs

```diff
@@ -1,16 +1,30 @@
 ﻿namespace Go.CodeAnalysis.Text
 {
+    /// <summary>
+    /// Abstract representation of a text segment.
+    /// </summary>
     public abstract class SnapshotBase
     {
         protected SnapshotBase(int length)
         {
             this.Length = length;
         }
 
+        /// <summary>
+        /// Gets the character located at a specific index in the text segment.
+        /// </summary>
+        /// <param name="offset"> Index of character in the text segment. </param>
+        /// <returns></returns>
         public abstract char this[int offset] { get; }
 
+        /// <summary>
+        /// Create an instance of concrete subclass. 
```

Comment:
This isn’t quite accurate, this property is getting a SnapshotSegment (a span within the document) that spans the length of the document. It’s still based on the abstract document, however.

Reply:
Yeah I was a bit confused by the pattern here. Why is base class dependent on subclass?

Reply:
Removing this doc for now. Will discuss separately.